File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder ans = new StringBuilder();
        int count = 0;
        char prev = '\0';
        
        for (char c : s.toCharArray()) {
            if (c == prev) {
                count++;
            } else {
                count = 1;
                prev = c;
            }
            if (count < 3) {
                ans.append(c);
            }
        }
        return ans.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string r;
        for (int i = 0; i < s.size(); i++)
            if (i < 2 || s[i] != s[i - 1] || s[i] != s[i - 2])
                r += s[i];
        return r;
    }
};
